services/user-service.js

```diff
@@ -0,0 +1,36 @@
+// use prisma
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+
+const bcrypt = require('bcryptjs');
+
+const userServices = {
+  signUp: async (req, cb) => {
+    try {
+      const { name, email, password } = req.body;
+      // find existed email
+      const user = await prisma.users.findUnique({ where: { email: email } });
+      if (user) throw new Error('email已經註冊過囉！');
+      if (!email.trim()) throw new Error('email為必填欄位！');
+      if (!name.trim()) throw new Error('名字為必填欄位！');
+      if (!password.trim()) throw new Error('密碼為必填欄位！');
+      // hash password
+      const hash = await bcrypt.hash(password, 10);
+      // create user
+      const postUser = await prisma.users.create({
+        data: {
+          email: email,
+          name: name,
+          password: hash,
+        },
+      });
+      // return info
+      const result = postUser;
+      delete result.password;
```

Comment:
wrap a model layer for this kind of stuff

Reply:
https://www.prisma.io/docs/concepts/components/prisma-client/select-fields

Reply:
option1: 
extractUser = (user: PrismaUser) => {
  const result = postUser;
      delete result.password;

  return postUser
}

Reply:
option2
```js
class UserModel {
  createUser(data) {
    
  }

  getUserById(id) {}
  getUsersByXXX() {}
}
```

Reply:
ok